lib/postgres_to_redshift.rb

```diff
@@ -128,14 +128,49 @@ def import_table(table)
     puts "Importing #{table.target_table_name}"
     target_connection.exec("DROP TABLE IF EXISTS public.#{table.target_table_name}_updating")
 
-    target_connection.exec("BEGIN;")
+    begin
+      target_connection.exec('BEGIN;')
+
+      target_connection.exec("ALTER TABLE public.#{target_connection.quote_ident(table.target_table_name)} RENAME TO #{table.target_table_name}_updating")
+
+      target_connection.exec("CREATE TABLE public.#{target_connection.quote_ident(table.target_table_name)} (#{table.columns_for_create})")
+
+      target_connection.exec("COPY public.#{target_connection.quote_ident(table.target_table_name)} FROM 's3://#{ENV['S3_DATABASE_EXPORT_BUCKET']}/export/#{table.target_table_name}.psv.gz' CREDENTIALS 'aws_access_key_id=#{ENV['S3_DATABASE_EXPORT_ID']};aws_secret_access_key=#{ENV['S3_DATABASE_EXPORT_KEY']}' GZIP TRUNCATECOLUMNS ESCAPE DELIMITER as '|';")
+
+      target_connection.exec('COMMIT;')
+
+    rescue PG::InternalError => exception
+      handle_pg_exception(table, exception)
+    end
+  end
 
-    target_connection.exec("ALTER TABLE public.#{target_connection.quote_ident(table.target_table_name)} RENAME TO #{table.target_table_name}_updating")
+  def handle_pg_exception(table, exception)
+    target_connection.exec('ROLLBACK;')
+
+    if exception.message.include?('stl_load_errors')
+      puts exception.message
+      puts "ERROR:  Last entry in Redshift's 'stl_load_errors' table:"
+      print_last_redshift_loading_error
+    else
+      puts 'ERROR:  Unhandled PG error:'
```

Comment:
I still think this should "raise", right? If we aren't explicitly handling the exception, then we need to ensure it errors correctly. Otherwise, we may hit this over and over again for something we are not handling.